### PR TITLE
[pdpix] Don't Panic on System Calls

### PR DESCRIPTION
**Pull request by ppenna**
Description
========

In this PR I fix issue https://github.com/demikernel/demikernel/issues/94.

Additionally, note also that I added dummy calls for the `setsockopt()` and `getsockopt()` system calls that shall be introduced in upcoming versions of Demikernel.

Related Issues
==========

- https://github.com/demikernel/demikernel/issues/72
- https://github.com/demikernel/demikernel/issues/73
- https://github.com/demikernel/demikernel/issues/74

### CODE REVIEW

File: src/demikernel/bindings.rs [hunk 1]
```diff
@@ -332,20 +451,76 @@ pub extern "C" fn dmtr_sgaalloc(size: libc::size_t) -> dmtr_sgarray_t {
...
+
+//==============================================================================
+// Standalone Functions
+//==============================================================================
+
+/// Converts a [sockaddr] into a [Ipv4Endpoint].
+fn sockaddr_to_ipv4endpoint(saddr: *const sockaddr) -> Result<Ipv4Endpoint, Fail> {
+    let sin: libc::sockaddr_in =
+        unsafe { *mem::transmute::<*const sockaddr, *const libc::sockaddr_in>(saddr) };
+    let addr: Ipv4Addr = { Ipv4Addr::from(u32::from_be_bytes(sin.sin_addr.s_addr.to_le_bytes())) };
```
Comment (BrianZill) [inline src/demikernel/bindings.rs:523]: What is going on here with byte order?  To the naive reader (i.e. me :-)) this looks like s_addr is being taken as little endian and being read as big endian.

Comment (ppenna) [reply src/demikernel/bindings.rs:523]: I did not author this and I also have no clue why the person intentionally did it.

I just noticed that the same code was everywhere, so I at least created a function to do this copy-and-paste job.

I'm happy to take any improvements on this function afterwards, but considering its name, please let me know what your suggestion is.

Comment (BrianZill) [reply src/demikernel/bindings.rs:523]: Okay, mark it with a ToDo comment to review it later, file an Issue about it, and we'll leave it as is for now.  My guess is that whoever wrote this didn't understand what was going on and just kept putting in byte-order conversions until it worked :-).

**Review by BrianZill — APPROVED** [review]
Looks good now.

File: src/demikernel/bindings.rs (hunk 1 above)
Comment (ppenna) [reply src/demikernel/bindings.rs:524]: Sure. I created issue https://github.com/demikernel/demikernel/issues/101 to keep track of this.

**Pull request merged**